src/algebra/free_algebra.lean

```diff
@@ -295,4 +295,43 @@ begin
 end
 (by { ext, simp, })
 
+
+open_locale big_operators
+
+/--
+Separate an element of the free algebra into its ℕ-graded components.
+
+This is defined as an `alg_hom` to `add_monoid_algebra` so that algebra operators can be moved
+before and after the mapping.
+-/
+noncomputable
+def grades : (free_algebra R X) →ₐ[R] add_monoid_algebra (free_algebra R X) ℕ :=
+lift R $ λ x, finsupp.single 1 $ ι R x
+
+/-- Recombining the grades recovers the original element-/
+lemma sum_id_grades :
+  (add_monoid_algebra.sum_id R).comp grades = alg_hom.id R (free_algebra R X) :=
+begin
+  ext,
+  unfold grades add_monoid_algebra.sum_id,
+  simp [finsupp.sum_single_index],
+end
+
+noncomputable
+instance : has_coe (add_monoid_algebra (free_algebra R X) ℕ) (free_algebra R X) := ⟨
+  (add_monoid_algebra.sum_id R : add_monoid_algebra (free_algebra R X) ℕ →ₐ[R] (free_algebra R X))
+⟩
+
+@[simp, norm_cast]
+lemma coe_def (x : add_monoid_algebra (free_algebra R X) ℕ) : (x : free_algebra R X) = add_monoid_algebra.sum_id R x := rfl
```

Comment:
This seemed harmless, and it makes the mouthful that is `add_monoid_algebra.sum_id` not really matter

Reply:
You mean the instance? Why not make it a local instance?